### ISSUE TITLE
medusa + Python API: inconsistent structure alignment

### ISSUE THREAD

**Issue by ahigerd** [mgba-emu/mgba#1861]
Unfortunately, Python CFFI doesn't support the align attribute, so `DSMemory`'s use of `ATTRIBUTE_ALIGN` causes the Python API to break when trying to use it for DS:

```
ffi.error: struct DSMemory: wrong offset for field 'io9' (cdef says 1398, but C compiler says 1400). fix it or use "...;" as the last field in the cdef for struct DSMemory to make it flexible
```

As a workaround I was able to insert a `uint16_t padding;` element in between `io7` and `io9` to manually align it.

**Comment by endrift** [comment]
Just extending the io7 array by one would probably be enough here too.

**Comment by ahigerd** [comment]
Yeah, that was the first thing I did but I didn't know what (if anything) that would imply semantically for that particular data structure.
